### ISSUE TITLE
People: Make face recognition results independent of thumbnail size

### ISSUE THREAD

**Issue by alexislefebvre** [photoprism/photoprism#2506]
**Is your feature request related to a problem? Please describe.**

Configuring the size of thumbnails can affect the face recognition negatively.

This was added as a warning:

> Reducing the *Static Size Limit* of thumbnails has a **significant impact on  [facial recognition](../user-guide/organize/people.md)     and image classification** results. Simply put, it means that the indexer can no longer see properly.

[source](https://github.com/photoprism/photoprism-docs/commit/4fa3bb0679100905e38b9f230f6304059ebddebd)

**Describe the solution you'd like**

Perform face recognition on : 
- original images
- or if it would too heavy for the indexer, generate thumbnails for the face recognition automatically, whatever the the size of thumbnails is; in other words, allow the indexer to generate thumbnails for the facial recognition, not too small or too big, to allow the best recognition without wasting disk storage

**Describe alternatives you've considered**

-

**Additional context**

-

**Comment by lastzero** [comment]
We could automatically increase the minimum thumb size if face recognition is enabled, but that may conflict with the config options. Just like on demand generation may conflict. IMHO it's best not to tinker too much and keep the defaults.

**Comment by alexislefebvre** [comment]
Fair enough, I used a [trick to reduce the generation of thumbnails](https://github.com/photoprism/photoprism-docs/commit/a49228888382bf1d09cc79865fa0c66b951e4816) but now I will favour the facial recognition. I'll restore the thumbnail sizes to default values and see how it works. And #1874 will avoid redundant thumbnails.

**Comment by gyto6** [comment]
If we could use the original instead of the thumbnail, it could be better for users (as me) and profesionnals, with a dedicated server.

Furthermore, with the Pull [#2478](https://github.com/photoprism/photoprism/pull/2478), it could be better for external client to access the originals instead of thumbnails. 

Thumbnails is great for Photoprism as this is one of the tricks that make it popular, but some companies or sponsors might prefers access to a picture with its full definition.

**Comment by lastzero** [comment]
TensorFlow models expects specific crops, you can't just use the original. Pure detection without recognizing on an original might work, but will be super slow and the "faces" it might find will be tiny on a regular screen.

**Comment by gyto6** [comment]
You know better than me. 

A simple suggestion as Deepstack works fine with original files for detection and recognition as far that I tested it. 

I'm simply stuck by now with many faces not detected or false positive and it might be another option instead of waiting for TF2 integration with/without new models and with/without GPU acceleration.

Thanks again whatever, I'll already wouldn't have classed my pictures this well without your solution.

**Comment by Rathna-K** [comment]
> We could automatically increase the minimum thumb size if face recognition is enabled, but that may conflict with the config options. Just like on demand generation may conflict. IMHO it's best not to tinker too much and keep the defaults.

can this be done on the Original and limit the number of thumbnail generation? with 350GB originals, I am going to have significant storage added for thumbs alone...
